### ISSUE TITLE
Multithread sim test failed bc one node cannot reach justified checkpoint

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#2197]
**Describe the bug**

3 nodes reach justified checkpoint and stop while 1 node can't reach it and keep running

**Expected behavior**

+ The test passes

**Steps to Reproduce**

+ it mainly happens in CI environment https://github.com/ChainSafe/lodestar/pull/2194/checks?check_run_id=2138682165

Log
[failed_multiThead.txt](https://github.com/ChainSafe/lodestar/files/6163767/failed_multiThead.txt)

**Comment by twoeths** [comment]
Node 0 has a different block 41 while the other 3 have same block 41
```
2021-03-18 10:26:00 [NODE 0]           info: Synced - finalized: 0 0x0000…0000 - head: 41 0x1c54…07f5 - peers: 3
2021-03-18 10:26:00 [NODE 1]           info: Synced - finalized: 0 0x0000…0000 - head: 41 0x614b…5952 - peers: 3
2021-03-18 10:26:00 [NODE 2]           info: Synced - finalized: 0 0x0000…0000 - head: 41 0x614b…5952 - peers: 3
2021-03-18 10:26:00 [NODE 3]           info: Synced - finalized: 0 0x0000…0000 - head: 41 0x614b…5952 - peers: 3
```

**Comment by twoeths** [comment]
this is interesting, 2 validators proposed 2 blocks at the same time:
```
2021-03-18 10:25:59 [VALI 0-7]         info: Validator is proposer! slot=41, validator=0xb89bebc699769726a318c8e9971bd3171297c61aea4a6578a7a4f94b547dcba5bac16a89108b6b6a1fe3695d1a874a0b
2021-03-18 10:25:59 [VALI 8-15]        info: Validator is proposer! slot=41, validator=0x876dd4705157eb66dc71bc2e07fb151ea53e1a62a0bb980a7ce72d15f58944a8a3752d754f52f4a60dbfc7b18169f268
2021-03-18 10:25:59 [VALI 0-7]         info: Proposed block hash=0x1c549d2f2c2d7ccff2dc6c0c01bc345409d9449d69068258fb7f913cbdaf07f5, slot=41
2021-03-18 10:25:59 [VALI 8-15]        info: Proposed block hash=0x614bd8ec8ecb056a8c160cfdb207bc9238551e4d5b284edf2175e60806525952, slot=41
```

this is bc node 0 had a different chain right before that time
```
2021-03-18 10:25:58 [NODE 3]           info: Synced - finalized: 0 0x0000…0000 - head: 30 0x8d9a…6774 - peers: 3
2021-03-18 10:25:58 [NODE 2]           info: Synced - finalized: 0 0x0000…0000 - head: 30 0x8d9a…6774 - peers: 3
2021-03-18 10:25:58 [NODE 0]           info: Synced - finalized: 0 0x0000…0000 - head: 25 0x5324…2b6a - peers: 3
2021-03-18 10:25:58 [NODE 1]           info: Synced - finalized: 0 0x0000…0000 - head: 30 0x8d9a…6774 - peers: 3
```

**Comment by dapplion** [comment]
Another run with failed sim tests
[9_Simulation multi thread multi node test.txt](https://github.com/ChainSafe/lodestar/files/6163968/9_Simulation.multi.thread.multi.node.test.txt)

**Comment by twoeths** [comment]
so node 0 had a completely different chain from genesis, this is because all nodes only completed initial sync after slot 7 and gossip only started after that while validator started proposing block from slot 5.

I think the correct behavior atm would be all nodes completed its initial sync at genesis (that's what the old initial sync work) so gossip could be started right away and all nodes could receive same blocks.

@dapplion how do you think this work in your new sync work? and is it possible to bypass the initial sync at genesis?

**Comment by dapplion** [comment]
@tuyennhv  The `range-sync` branch features this state getter with some additional rules that the current sync doesn't have. It is able to realize that if you are really close to the clock slot you are synced, and doesn't trigger an initial sync
https://github.com/ChainSafe/lodestar/blob/629f88017498e7ca5e41d5fd60f88358da9765b6/packages/lodestar/src/sync/sync.ts#L97-L120

We can port this bit of logic to master

**Comment by dapplion** [comment]
This issue will be partially fixed with https://github.com/ChainSafe/lodestar/pull/2212

> so node 0 had a completely different chain from genesis, this is because all nodes only completed initial sync after slot 7 and gossip only started after that while validator started proposing block from slot 5.

@tuyennhv Even if this happens shouldn't the forkchoice of node 0 pick the correct chain eventually and re-org?

**Comment by twoeths** [comment]
> This issue will be partially fixed with #2212
> 
> > so node 0 had a completely different chain from genesis, this is because all nodes only completed initial sync after slot 7 and gossip only started after that while validator started proposing block from slot 5.
> 
> @tuyennhv Even if this happens shouldn't the forkchoice of node 0 pick the correct chain eventually and re-org?

after slot 7 these 2 chains receive blocks from each other but they could not process due to `BLOCK_ERROR_PARENT_UNKNOWN` and then could not add to forkchoice, eventually 3 nodes stop and node 1 run until timeout. What I don't know is why `onUnknownBlockRoot` did not work to handle that error.

**Comment by dapplion** [comment]
We've added logic to turn on gossip on genesis. To debug the `onUnknownBlockRoot` issue see https://github.com/ChainSafe/lodestar/issues/2409
